### ISSUE TITLE
Implement Filtering by Name, Address, Alt #, Building Permit, Created By from Column Headers on My Project Page

### ISSUE THREAD

**Issue by entrotech** [hackforla/tdm-calculator#1647]
### Overview
- Implement the Name, Address, Alt #, Building Permit, Created By Filters on the My Projects Page per the Figma Prototype references in the Resources paragraph below to comply with the filter re-design of Issue #1578. 

### Action Items
- [x] This is partially implemented by Issue #1801. Please read that issue first for background information, adjustments John Darragh made to the design of all the filter pop-ups for consistency with our style guide, and information about what is done so far.
- [x] For this issue, you will need to complete the <TextPopup> component implementation.  You will probably need to modify one or more parent components to pass an additional prop that references the entire collection of projects, so you can extract the list of matches for the "auto-complete" component the design calls for.
- [x] Use the <DatePopup> component implementation in the same directory as an example.
- [x] remove the dependency from #1649 and move it to new issue approval column.
 
### Resources/Instructions
Refer to this Figma prototype: https://www.figma.com/proto/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?type=design&node-id=12173-2052&t=wQuxLI1JbOw0OzLn-1&scaling=scale-down&page-id=2499%3A11472&starting-point-node-id=11997%3A885

**Comment by roslynwythe** [comment]
Progress:
- Filtering on `Name`, `Address`, `Alt #` and `Created By` columns is functional

<details><summary>Details</summary>

![image](https://github.com/user-attachments/assets/c1a2083c-002d-4c5b-b108-5fa278628622)

</details> 
Questions:
- I don't see `Building Permit` column so I'm not sure how to implement that filter
- Using the Figma link under Resources, I was not able to find a prototype of the filter autocomplete UI to answer questions such as:
   - how many options should be displayed (or how much vertical space should be allocated) for the dropdown?
   - should the options populate in the dropdown immediately or asynchronously (for instance, after the user types the first character?)
   - Other UI issues were raised in the meeting:
      - Should there be a visual cue that a column is being filtered (perhaps the filter icon should change?)
      - Should there be a way to clear all column filters?

**Comment by entrotech** [comment]
Yes, the building permit column is gone, so we don't need that.

Others have encountered the problem with the pre-commit hook as well.  It's a  problem with an npm package called lerna, which allows you to run commands on multiple projects in a monorepo with a single command - in this case the pre-commit hook tries to run lint on the client and server projects.  The pre-commit hook tries to run the lerna command from the command line and is failing. Here's something to try: Open a terminal window and set the directory to the root folder of the project. Then run
``` npm install -g lerna```
from the terminal to install lerna globally on your machine.
Let me know it this works - if it does, I'll add it to the Contributing Code page of the wiki.

If the commit still doesn't work, you can work-around it by committing with the --no-verify option to bypass the pre-commit hook:
```git commit --no-verify -m "<your commit message>"```

(But then we still need to figure out what needs to be done to get the hook to work.)

Also, you will probably find that the ,/push.sh script won't work, since there are some failing unit tests that need to get fixed, so you will need to use the alternative 
```git push origin HEAD```

I look forward to checking out your PR ;-)

**Comment by roslynwythe** [comment]
Thank you @entrotech - running the command `npm install -g lerna` resolved the pre-commit problem.  Earlier in the day, I suspected that the dependencies should be installed globally so I had tried `npm install -g` but that did not work.  

I have another question.  I am working on placing the search icon to the left of the placeholder text in the search input, and I was wondering, should I use a Material Design search icon or should I use the same icon that you are currently using in the search input?
